### PR TITLE
Get tests to pass in new TermInterface

### PR DESCRIPTION
**Pull request by nmheim**
@0x0f0f0f here are my fixes for #584.

### CODE REVIEW

File: src/types.jl [hunk 1]
```diff
@@ -136,6 +136,7 @@ function arguments(x::BasicSymbolic)
     end
     return args
 end
+children(x::BasicSymbolic) = arguments(x)
```
Comment (nmheim) [inline src/types.jl:139]: this could be fishy

File: src/utils.jl [hunk 2]
```diff
@@ -221,7 +221,7 @@ macro matchable(expr)
         $expr
         SymbolicUtils.iscall(::$name) = true
         SymbolicUtils.operation(::$name) = $name
-        SymbolicUtils.arguments(x::$name) = getfield.((x,), ($(QuoteNode.(fields)...),))
+        SymbolicUtils.children(x::$name) = getfield.((x,), ($(QuoteNode.(fields)...),))
```
Comment (nmheim) [inline src/utils.jl:224]: as well as this

File: src/types.jl [hunk 3]
```diff
@@ -813,7 +816,7 @@ function show_term(io::IO, t)
         show_pow(io, args)
     elseif f === (getindex)
         show_ref(io, f, args)
-    elseif f === (identity) && !issym(args[1]) && iscall(args[1])
+    elseif f === (identity) && !issym(args[1]) && !iscall(args[1])
```
Comment (0x0f0f0f) [inline src/types.jl:819]: GOOD CATCH

File: src/types.jl (hunk 1 above)
Comment (nmheim) [reply src/types.jl:139]: this was what I was thinking as well, but then stumbled over `arguments` calling `unsorted_arguments` which is part of TI, so I kept it the current way. should we rename `unsorted_arguments` as well then? 🤔

Comment (0x0f0f0f) [reply src/types.jl:139]: Ok, sounds good for now. Maybe we should file an issue to TI directly. cc @shashi

File: src/utils.jl (hunk 2 above)
Comment (nmheim) [reply src/utils.jl:224]: ahhh, yes, that makes sense. missed the `iscall(::$name) = true` 👍

Comment (0x0f0f0f) [reply src/utils.jl:224]: Also missing `head`

File: src/types.jl (hunk 1 above)
Comment (shashi) [reply src/types.jl:139]: Shouldnt children also include the operation at the beginning?

**Pull request merged**